controller/backup_controller.go

```diff
@@ -278,6 +291,11 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 				return nil // Ignore error to prevent enqueue
 			}
 
+			if err := bc.isBackupBeingUsedForVolumeRestore(backup.Name, backupVolumeName); err != nil {
+				log.WithError(err).Warnf("Unable to delete remote backup %v", backup.Name)
+				return err
+			}
```

Comment:
Should we return err to requeue? just noticed the queue for backup is always resync in a periodical time by default. I think we should retry, but I am also curious why we have periodical resync for the queue. 

```go
	ds.BackupInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
		AddFunc:    bc.enqueueBackup,
		UpdateFunc: func(old, cur interface{}) { bc.enqueueBackup(cur) },
		DeleteFunc: bc.enqueueBackup,
	})
```

Reply:
The resync period is 30 seconds. We can rely on the timer instead of the retries by error.

Reply:
So do we still need to return error? I saw the code is the same.

Reply:
Ah..I forgot push the latest code...

Reply:
Ah thanks. I also missed the check 😅